[IDE][Sema] Add test cases for issues no longer occurring in the stress tester

ahoppen opened the pull request:
SR-14708 and SR-15497 are no longer happening in the stress tester. Add their test case to the test suite.

rdar://78781625
rdar://85602946

ahoppen commented:
@swift-ci Please smoke test

xedin commented on validation-test/compiler_crashers_2_fixed/sr15497.swift:
```diff
@@ -0,0 +1,12 @@
+// RUN: %swift-frontend -typecheck  %s
+
+public struct {
```
Forgot the name here.

ahoppen replied:
Not including a name was part of the test case in the bug report, so it might have been crucial to trigger the crash before. So: This is intentional.

xedin replied:
Okay, you need to add `not`  before %swift-frontend then if you don't want to deal with error messages.

ahoppen commented:
@swift-ci Please smoke test

slavapestov commented on validation-test/compiler_crashers_2_fixed/sr15497.swift:
```diff
@@ -0,0 +1,12 @@
+// RUN: not %swift-frontend -typecheck  %s
```
%target-swift-frontend

slavapestov commented on validation-test/IDE/stress_tester_issues_fixed/sr14708.swift:
```diff
@@ -0,0 +1,34 @@
+// RUN: %swift-ide-test --code-completion --source-filename %s --code-completion-token=COMPLETE | %FileCheck %s
```
%target-swift-ide-test

ahoppen committed:
[IDE] Add test case for SR-14708

SR-14708 is no longer happening in the stress tester. Add its test case to the test suite.

rdar://78781625

ahoppen committed:
[Sema] Add test case for SR-15497

SR-15497 is no longer happening in the stress tester. Add its test case to the test suite.

rdar://85602946

ahoppen commented:
@swift-ci Please test
